The Last Release 2.0.502 Broke My Ability to Jack-In -- repl connection fails

This just happened today after update in Cursor 0.49.6. From the developers tool console:

[betterthantomorrow.calva]Expected repl window to have a namespace!
$onExtensionRuntimeError @ workbench.desktop.main.js:2489
S @ workbench.desktop.main.js:660
Q @ workbench.desktop.main.js:660
M @ workbench.desktop.main.js:660
L @ workbench.desktop.main.js:660
(anonymous) @ workbench.desktop.main.js:660
B @ workbench.desktop.main.js:46
fire @ workbench.desktop.main.js:46
fire @ workbench.desktop.main.js:2492
l.onmessage @ workbench.desktop.main.js:4938
workbench.desktop.main.js:2489 Error: Expected repl window to have a namespace!
	at e.assertIsDefined (/home/dsearsmith/.cursor/extensions/betterthantomorrow.calva-2.0.502/out/extension.js:2:987809)
	at Object.c [as getNamespace] (/home/dsearsmith/.cursor/extensions/betterthantomorrow.calva-2.0.502/out/extension.js:2:821230)
	at Timeout._onTimeout (/home/dsearsmith/.cursor/extensions/betterthantomorrow.calva-2.0.502/out/extension.js:2:995718)
	at listOnTimeout (node:internal/timers:581:17)
	at process.processTimers (node:internal/timers:519:7)
$onExtensionRuntimeError @ workbench.desktop.main.js:2489
S @ workbench.desktop.main.js:660
Q @ workbench.desktop.main.js:660
M @ workbench.desktop.main.js:660
L @ workbench.desktop.main.js:660
(anonymous) @ workbench.desktop.main.js:660
B @ workbench.desktop.main.js:46
fire @ workbench.desktop.main.js:46
fire @ workbench.desktop.main.js:2492
l.onmessage @ workbench.desktop.main.js:4938
workbench.desktop.main.js:2479 [Extension Host] Failed connecting: TypeError: Cannot read properties of undefined (reading 'replType')
	at Object.update (/home/dsearsmith/.cursor/extensions/betterthantomorrow.calva-2.0.502/out/extension.js:2:964678)
	at Object.update (/home/dsearsmith/.cursor/extensions/betterthantomorrow.calva-2.0.502/out/extension.js:2:962817)
	at /home/dsearsmith/.cursor/extensions/betterthantomorrow.calva-2.0.502/out/extension.js:2:634186
	at Generator.next (<anonymous>)
	at u (/home/dsearsmith/.cursor/extensions/betterthantomorrow.calva-2.0.502/out/extension.js:2:632773)
	at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
s_r @ workbench.desktop.main.js:2479
$logExtensionHostMessage @ workbench.desktop.main.js:2479
S @ workbench.desktop.main.js:660
Q @ workbench.desktop.main.js:660
M @ workbench.desktop.main.js:660
L @ workbench.desktop.main.js:660
(anonymous) @ workbench.desktop.main.js:660
B @ workbench.desktop.main.js:46
fire @ workbench.desktop.main.js:46
fire @ workbench.desktop.main.js:2492
l.onmessage @ workbench.desktop.main.js:4938"

I reverted to the previous version and all is better.

If it helps, AI said "Calva telling us that it has no record of a live REPL window with a namespace prompt, so when it tries to run its session-initialization logic it blows up. In practice that means Calva never “attached” its REPL UI pane to the nREPL session you manually reconnected." Not sure that makes any sense.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,8 @@ Changes to Calva.
 
 ## [Unreleased]
 
+- Fix: [Release 2.0.502 repl connection fails, breaking Jack-In](https://github.com/BetterThanTomorrow/calva/issues/2799) (Hopefully)
+
 ## [2.0.502] - 2025-04-26
 
 - Fix: [formatCode throws an error when output output is printed in the output window](https://github.com/BetterThanTomorrow/calva/issues/2791)

--- a/src/nrepl/repl-session.ts
+++ b/src/nrepl/repl-session.ts
@@ -36,7 +36,8 @@ function getSession(fileType?: string): NReplSession {
   ) {
     return outputWindow.getSession();
   } else {
-    return cljsLib.getStateValue(sessionKey);
+    const session = cljsLib.getStateValue(sessionKey);
+    return session || null; // Return null instead of undefined when session doesn't exist
   }
 }
 

